Sources/Brave/Migration/Migration.swift

```diff
@@ -122,16 +122,25 @@ public class Migration {
         let historySnapshot = oldTab.urlHistorySnapshot as? [String] ?? []
         
         for url in historySnapshot {
-          guard let url = NSURL(idnString: url) as? URL ?? URL(string: url) else { continue }
+          guard let url = NSURL(idnString: url) as? URL ?? URL(string: url) else {
+            Logger.module.error("Failed to parse URL: \(url) during Migration!")
```

Comment:
if this is never mean to happen we can consider adding assertion failure here

Reply:
No lets not add that please, I have no idea how the previous problem happened. It will be better not to have a random crash for an reason which is hard to explain.

Reply:
This would help us catch this problem though? assertion failure is not going to crash outside running it locally

Reply:
Is there an ability to send crash report without crashing? we do this in brave-core (`DumpWithoutCrashing`)